### PR TITLE
2117: replace password confirmation with password visibility toggle

### PR DESCRIPTION
**Pull request by kpethtel**
**Story card:** https://app.clubhouse.io/simpledotorg/story/2117

## Because

This is part of an epic that involves changing our password requirements to a checklist that goes green as the requirements are met. The confirmation process doesn't suit that design well, but removing the confirmation necessitates a visibility toggle. 

## This addresses

These are entirely front end changes that improve UI and move us toward a new design.

## Testing

1) Test the new invitations page.
In Simple server, go to the More tab then down to Dashboard Admin. Invite a new user
Get the invitation email either from the rails logs or mail catcher. 
Follow the link and test for the conditions listed below.

2) Test the email reset page
At the login page, click "forgot my password" and enter your email
Get the invitation email either from the rails logs or mail catcher. 
Follow the link and test for the conditions listed below.

Tests for each page:
- Confirm that the toggle hides and unhides as expected.
- Confirm that appearance is as expected (this primarily applies to Claudio)
- Confirm that incorrect passwords fail and present an error message
- Confirm that correct passwords succeed

### CODE REVIEW

File: app/views/email_authentications/passwords/edit.html.erb [hunk 1]
```diff
@@ -16,3 +16,7 @@
 <div class="actions mt-4">
   <%= render "email_authentications/shared/links" %>
 </div>
+
+<script>
+  new PasswordVisibilityToggle
```
Comment (rsanheim) [inline app/views/email_authentications/passwords/edit.html.erb:21]: minor thing, but can you throw parens at the end of this function call?

```
new PasswordVisibilityToggle();
```

File: app/views/email_authentications/invitations/edit.html.erb [hunk 2]
```diff
@@ -4,13 +4,15 @@
...
+          <%= f.password_field :password, type: "password", id: "password", class: "password-field", autofocus: true, autocomplete: "off" %>
+          <input id="toggle-password" type="button" class="btn btn-light password-toggle-button" value="Show">
+        </div>
       <%= f.primary t("devise.invitations.edit.submit_button") %>
     <% end %>
   </div>
 </div>
+
+<script>
+  new PasswordVisibilityToggle
```
Comment (rsanheim) [inline app/views/email_authentications/invitations/edit.html.erb:17]: parens and closing semicolon here as well

**Review by rsanheim — APPROVED** [review]
some minor JS feedback, otherwise LGTM

File: app/views/email_authentications/passwords/edit.html.erb (hunk 1 above)
Comment (kpethtel) [reply app/views/email_authentications/passwords/edit.html.erb:21]: Good call. Thanks. Something felt off about that. Most of my JS experience is in frameworks, so I sometimes feel out of my element in vanilla JS.

**Pull request merged**